tools/interop_matrix/run_interop_matrix_tests.py

```diff
@@ -0,0 +1,186 @@
+argp.add_argument('-j', '--jobs', default=multiprocessing.cpu_count(), type=int)
+argp.add_argument('--gcr_path',
+                  default='gcr.io/grpc-testing',
+                  help='Path of docker images in Google Container Registry')
+
+argp.add_argument('--release',
+                  default='master',
+                  choices=['all', 'master'] + _RELEASES,
+                  help='Release tags to test.  When testing all '
+                  'releases defined in client_matrix.py, use "all".')
+
+argp.add_argument('-l', '--language',
+                  choices=['all'] + sorted(_LANGUAGES),
+                  nargs='+',
+                  default=['all'],
+                  help='Languages to test')
+
+argp.add_argument('--keep',
+                  action='store_true',
+                  help='keep the created local images after finishing the tests.')
+
+argp.add_argument('--report_file',
+                  default='report.xml',
+                  help='The result file to create.')
+
+args = argp.parse_args()
+
+
+def find_all_images_for_lang(lang):
+  """Find docker images for a language across releases and runtimes.
+
+  Returns dictionary of list of (<tag>, <image-full-path>) keyed by runtime.
+  """
+  # Find all defined releases.
+  if args.release == 'all':
+    releases = client_matrix.LANG_RELEASE_MATRIX[lang]
+  else:
+    # Look for a particular release.
+    if args.release not in ['master'] + client_matrix.LANG_RELEASE_MATRIX[lang]:
+      jobset.message('SKIPPED',
+                     '%s for %s is not defined' % (args.release, lang),
+                     do_newline=True)
+      return []
+    releases = [args.release]
+
+  # Images tuples keyed by runtime.
+  images = {}
+  for runtime in client_matrix.LANG_RUNTIME_MATRIX[lang]:
+    image_path = '%s/grpc_interop_%s' % (args.gcr_path, runtime)
+    output = subprocess.check_output(['gcloud', 'beta', 'container', 'images',
+                                        'list-tags', '--format=json', image_path])
+    docker_image_list = json.loads(output)
+    # All images should have a single tag or no tag.
+    tags = [i['tags'][0] for i in docker_image_list if i['tags']]
+    jobset.message('START', 'Found images for %s: %s' % (image_path, tags),
+                   do_newline=True)
+    skipped = len(docker_image_list) - len(tags)
+    jobset.message('START', 'Skipped images (no-tag/unknown-tag): %d' % skipped,
+                   do_newline=True)
+    # Filter tags based on the releases.
+    images[runtime] = [(tag,'%s:%s' % (image_path,tag)) for tag in tags if
+                       tag in releases]
+  return images
+
+# caches test cases (list of JobSpec) loaded from file.  Keyed by lang and runtime.
+_loaded_testcases = {}
+def find_test_cases(lang, release):
+  """Returns the list of test cases from testcase files per lang/release."""
+  file_tmpl = os.path.join(os.path.dirname(__file__), 'testcases/%s__%s')
```

Comment:
I'd like to reserver single _ in case the we need to use it inside the release tag or language.